### ISSUE TITLE
Only load AppCache in production

### ISSUE THREAD

**Issue by mxstbr** [mxstbr/react-boilerplate#90]
Currently, AppCache is loaded in both production and development, which leads to some funky scenarios in development sometimes.

The problem is that I have yet to figure out a way to add the `manifest` attribute to the `html` tag in the `index.html` file only in production. Any ideas?

**Comment by jantimon** [comment]
Just remove the AppCachePlugin for dev?
```
  plugins.push(new AppCachePlugin({ // AppCache should be in both prod and dev env
    exclude: ['.htaccess'] // No need to cache that. See https://support.hostgator.com/articles/403-forbidden-or-no-permission-to-access
  }));
```

**Comment by jantimon** [comment]
The html plugin will add the manifest for you:
See
https://github.com/ampedandwired/html-webpack-plugin/blob/feature/loaders/index.js#L357-L359
and
https://github.com/ampedandwired/html-webpack-plugin/blob/feature/loaders/index.js#L472-L481

**Comment by mxstbr** [comment]
I know that works, but then you get an error in development mode. The index.html looks like this:

```HTML
<!-- index.html -->
<html lang="en" manifest="manifest.appcache">
```

If we remove the AppCachePlugin for dev, the `manifest` attribute will still exist and it will throw an error.

**Comment by mxstbr** [comment]
Oh damn nice, that's exactly what I need! Thanks @jantimon, will look into it asap!

**Comment by jantimon** [comment]
Just remove `manifest="manifest.appcache"` and it should work (if not we have to fix the html-webpack-plugin)

**Comment by mxstbr** [comment]
Doesn't work...

I removed the manifest attribute, but it didn't get added in production.

**Comment by jantimon** [comment]
K I'll try to create an example in the html-webpack-plugin project

**Comment by mxstbr** [comment]
Since I use htmlWebpackPlugin in both dev and prod, wouldn't it add the manifest in both scenarios?

**Comment by jantimon** [comment]
No - it checks the current assets.
https://github.com/ampedandwired/html-webpack-plugin/tree/feature/loaders/examples/appcache

It's important to add the appcache plugin before the html plugin as they use the same plugin hook `emit`.

**Comment by mxstbr** [comment]
Nice, that worked, thanks so much!

Closed in 61740a6
